src/Util/MIMETypeUtil.m

```diff
@@ -1546,18 +1561,1023 @@ + (NSDictionary *)genericMIMETypesToExtensionTypes
+            @"xif" : @"image/vnd.xiff",
+            @"xla" : @"application/vnd.ms-excel",
+            @"xlam" : @"application/vnd.ms-excel.addin.macroenabled.12",
+            @"xlc" : @"application/vnd.ms-excel",
+            @"xlf" : @"application/x-xliff+xml",
+            @"xlm" : @"application/vnd.ms-excel",
+            @"xls" : @"application/vnd.ms-excel",
+            @"xlsb" : @"application/vnd.ms-excel.sheet.binary.macroenabled.12",
+            @"xlsm" : @"application/vnd.ms-excel.sheet.macroenabled.12",
+            @"xlsx" : @"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+            @"xlt" : @"application/vnd.ms-excel",
+            @"xltm" : @"application/vnd.ms-excel.template.macroenabled.12",
+            @"xltx" : @"application/vnd.openxmlformats-officedocument.spreadsheetml.template",
+            @"xlw" : @"application/vnd.ms-excel",
+            @"xm" : @"audio/xm",
+            @"xml" : @"application/xml",
+            @"xo" : @"application/vnd.olpc-sugar",
+            @"xop" : @"application/xop+xml",
+            @"xpi" : @"application/x-xpinstall",
+            @"xpl" : @"application/xproc+xml",
+            @"xpm" : @"image/x-xpixmap",
+            @"xpr" : @"application/vnd.is-xpr",
+            @"xps" : @"application/vnd.ms-xpsdocument",
+            @"xpw" : @"application/vnd.intercon.formnet",
+            @"xpx" : @"application/vnd.intercon.formnet",
+            @"xsl" : @"application/xml",
+            @"xslt" : @"application/xslt+xml",
+            @"xsm" : @"application/vnd.syncml+xml",
+            @"xspf" : @"application/xspf+xml",
+            @"xul" : @"application/vnd.mozilla.xul+xml",
+            @"xvm" : @"application/xv+xml",
+            @"xvml" : @"application/xv+xml",
+            @"xwd" : @"image/x-xwindowdump",
+            @"xyz" : @"chemical/x-xyz",
+            @"xz" : @"application/x-xz",
+            @"yang" : @"application/yang",
+            @"yin" : @"application/yin+xml",
+            @"z1" : @"application/x-zmachine",
+            @"z2" : @"application/x-zmachine",
+            @"z3" : @"application/x-zmachine",
+            @"z4" : @"application/x-zmachine",
+            @"z5" : @"application/x-zmachine",
+            @"z6" : @"application/x-zmachine",
+            @"z7" : @"application/x-zmachine",
+            @"z8" : @"application/x-zmachine",
+            @"zaz" : @"application/vnd.zzazz.deck+xml",
+            @"zip" : @"application/zip",
+            @"zir" : @"application/vnd.zul",
+            @"zirz" : @"application/vnd.zul",
+            @"zmm" : @"application/vnd.handheld-entertainment+xml",
+        };
+    });
+    return result;
+}
+
 + (nullable NSString *)fileExtensionForMIMETypeViaLookup:(NSString *)mimeType
 {
     return [[self genericMIMETypesToExtensionTypes] objectForKey:mimeType];
 }
 
 + (nullable NSString *)fileExtensionForMIMEType:(NSString *)mimeType
 {
-    // Try to deduce the file extension by converting to a UTI type.
```

Comment:
This is an important change. Now that I've realized that the UTI type-based lookup for file extensions isn't 100% reliable, I'm going to prefer to determine the attachment file extension using lookup. This places a greater responsibility on our lookup mapping to be reliable.